### ISSUE TITLE
Unable to change checkbox variants to new SVG

### ISSUE THREAD

**Issue by BjornBirkelund** [ag-grid/ag-grid#7121]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Current behavior**

Checkbox icon variants aren't changing at all when a new svg is provided for the variant.
https://plnkr.co/edit/go8F2lwzZLvJda7B?open=index.jsx
I tried to keep it simple in this example - in lines 36 and 50 of styles.css I changed the link to point to the menu icon. 
The result is the icons don't change and just stay the same. 

In the documentation I can't find any examples where the checkbox svg is changed
https://www.ag-grid.com/react-data-grid/custom-icons/#svg-icons

**Expected behavior**
Checkbox icon can be changed by providing an SVG and targeting 
`.ag-theme-alpine .ag-icon-checkbox-checked {`
`.ag-theme-alpine .ag-icon-checkbox-indeterminate {`
and
.ag-theme-alpine .ag-icon-checkbox-unchecked {

Incase I am misinterpreting what this is targeting, in the below screenshot I am expecting the checkboxes on the left to change.
<img width="300" alt="Screenshot 2023-10-06 at 1 12 24 PM" src="https://github.com/ag-grid/ag-grid/assets/57355719/735a18fc-a5fa-402d-8217-9c8bc9baaf33">

**Comment by AG-Zoheil** [comment]
Hi @BjornBirkelund,

Thank you for submitting this on Github.

Let me clarify how and why our checkboxes work in a specific manner.

Our checkboxes are not icons in the same sense as we use icons around the grid elsewhere, because we actually render a real input field for accessibility purposes, and then cover that input with a CSS pseudo element. We don't add an ag-icon element to the DOM and because of that, there is no option to config them via JS.

For checkboxes, radio buttons and toggle inputs the only way to style them is via CSS as shown here: https://www.ag-grid.com/javascript-data-grid/custom-icons/#example-icons-alternative-font.

Please follow the above documentation as a guide for the styling.

Kind regards,
Zoheil

**Comment by BjornBirkelund** [comment]
Thank you for your reply, I attempted following that example from the doc, but it doesn't seem to specify anything for using custom svgs as opposed to an icon font , which is my usecase.

I have updated a code-sandbox showing a simple example combining the instructions from the doc and custom svgs, but they still don't update the icons. 
https://codesandbox.io/s/ag-grid-packages-forked-flzxz6?file=/src/styles.css

Thank you for your help.
